### DISCUSSION TITLE
Using Tedana with HCP pipelines?

### DISCUSSION THREAD

**Discussion by demiralsb** [ME-ICA/tedana#925] [Q&A]
Hello all,

I was wondering -- If I have a MEMB dataset (3 echoes, mb factor 5, Siemens Prisma 3T), and want to combine tedana with the HCP pipelines, what would be the best pipeline/denoising strategy?

CHeers,
Baris

**Comment by tsalo** [comment]
Hi Baris,

Do you know if the HCP pipelines can handle multi-echo data at all? The two main workflows we (the tedana devs) have worked with/on are AFNI (afni_proc.py) and fMRIPrep, which both have special steps to work well with multi-echo data. For example, motion correction should be performed using the same transforms across all echoes, which isn't something that most pipelines take into account. I'm not particularly familiar with the HCP pipelines, so I don't know if they do this.

Do the HCP pipelines have any features that make them particularly essential for you?

Best,
Taylor

**Reply by demiralsb** [reply]
Hi Taylor,

I do not see anything offered by HCP for multi-echo data preprocessing. We have been using HCP MB acquisition setup with HCP pipelines running in our servers -tradition-. 

However, when we started collecting MEMB data, I noticed that simple averaging/weighting of the echoes and entering this final .nii file to the MB HCP pipeline may not be a very optimal approach. Seeing tat tedana is using a sophisticated methodology I wanted to figure out how I can implement this. 

I am ready to be convinced that I can denoise and pre-process MEMB data as optimally as HCP pipeline can (assuming it is doing a great job). I do not have a very strong reason to stick with it.  

Open to your suggestions,

Thanks,
Baris

**Reply by tsalo** [reply]
Sorry for the slow follow-up. I am not very knowledgeable about the HCP pipeline, so I don't know if it does anything better or worse than fMRIPrep.

>However, when we started collecting MEMB data, I noticed that simple averaging/weighting of the echoes and entering this final .nii file to the MB HCP pipeline may not be a very optimal approach.

An unweighted average across echoes should be an improvement over single-echo fMRI (all else equal). You could try a PAID-based weighted average, which might be marginally better. I wouldn't recommend performing a T2*-based weighted average (which requires estimating T2* for each voxel) before motion correction, so it really is useful to have multi-echo support in the preprocessing pipeline.

If the HCP pipeline doesn't account for multi-echo data natively, then I think your best bet is to use something like AFNI or fMRIPrep, where steps like motion correction and distortion correction will be applied in the exact same manner across echoes.
